src/utils/fhirUtils.ts

```diff
@@ -0,0 +1,91 @@
+import { BasicMedicationRequestResource, BasicActionResource } from 'pathways-model';
+import { v1 } from 'uuid';
+
+// translates pathway recommendation resource into suitable FHIR resource
+export function translatePathwayRecommendation(
+  pathwayResource: BasicMedicationRequestResource | BasicActionResource,
+  patientId: string
+): fhir.Resource {
+  const { resourceType } = pathwayResource;
+  const resourceProperties = {
+    resourceType,
+    id: v1(),
+    intent: 'order',
+    subject: { reference: `Patient/${patientId}` },
+    status: 'active',
+    authoredOn: new Date().toISOString(),
+    meta: {
+      lastUpdated: new Date().toISOString()
+    }
+  };
+
+  switch (resourceType) {
+    case 'ServiceRequest': {
+      const { code } = pathwayResource as BasicActionResource;
+      return {
+        ...resourceProperties,
+        code
+      };
+    }
+    case 'MedicationRequest': {
+      const { medicationCodeableConcept } = pathwayResource as BasicMedicationRequestResource;
+      return {
+        ...resourceProperties,
+        medicationCodeableConcept
+      };
+    }
+    default: {
```

Comment:
This should probably throw an error rather than just returning an empty object. With the latest changes do we have any other types in the system?

Reply:
I think those 2 are currently the other resource types.  I pushed a commit to throw an error here.